en/_posts/2015-05-05-sailabove-documentation-predictor.markdown

```diff
@@ -38,15 +38,60 @@ That's all. All 3 containers are now handling their share of the requests on dom
 Now that you have a scalable, load-balanced application, you will most probably want to publish it with an appealing domain. Let's say ``www.sayhello.com``. All you need to do is:
 
 ```
-sail services domain-attach <user name>/hello-python www.sayhello.com
+$ sail services domain-attach <user name>/hello-python www.sayhello.com
 ```
 
 It can be detached at any time with:
 
 ```
-sail services domain-detach <user name>/hello-python www.sayhello.com
+$ sail services domain-detach <user name>/hello-python www.sayhello.com
 ```
 
+# Routing specific URLs to dedicated services
+
+URI based routing allows requests matching a specific pattern to be redirected to a dedicated service. For different applications, this allows to share a single domain under multiple prefixes. For micro-service oriented applications, this allows to deploy, monitor and scale each service endpoint independentely.
+
+For example, let's say we have both a blog and a redmine application and that they both should be accessible from a single, trusted domain. One could attached domains like:
+
+```
+$ sail services attach-domain <my-app>/wordpress interne.example.com --pattern /blog
+$ sail services attach-domain <my-app>/redmine   interne.example.com --pattern /redmine
+$ sail apps domain-list <my-app>
+DOMAIN               SERVICE             PATTERN
+interne.example.com  <my-app>/wordpress  /blog
+interne.example.com  <my-app>/redmine    /redmine
+```
+
+In this case, the predictor will route by prefix. In other words, all URLs starting by '/blog/' will go to wordpress and all requests starting with ''
+
+Most micro-service oriented architecture will require finer grained routing than prefixes. For this use case, patterns may be defined as simple regular expressions. Aa '*' between 2 consecutive '/' is a wildcard. It will match any value between the 2 consecutive '/'. This value may not itself contain a '/'. If a pattern ends with a '$', the pattern will act as a "mask" instead of a "prefix". This is basically the same meaning as '$' in regular expressions matching the end of the line.
+
+For example, let's build a trivial sms sending application:
+
+```
+/sms
+/sms/*
+/sms/*/send$
+```
+
+We'll attach the routes, like the blog+redmine example:
+
+```
+$ sail services attach-domain <my-app>/sms-dests api.sms.com --pattern '/sms'
+$ sail services attach-domain <my-app>/sms-list  api.sms.com --pattern '/sms/*'
```

Comment:
The final `*` is not needed, it might makes sense to remove it to emphasize that routes are actually prefixes when they are not terminated by a `$`.